packages/@ourworldindata/grapher/src/stackedCharts/StackedDiscreteBarChart.tsx

```diff
@@ -308,10 +323,13 @@ export class StackedDiscreteBarChart
                 sortByFunc = (item: Item): string => item.label
                 break
             case SortBy.column:
-                const sortColumnSlug = this.sortConfig.sortColumnSlug
-                sortByFunc = (item: Item): number =>
-                    item.bars.find((b) => b.columnSlug === sortColumnSlug)
-                        ?.point.value ?? 0
+                const owidRowsByEntityName =
+                    this.sortColumn?.owidRowsByEntityName
+                sortByFunc = (item: Item): number => {
+                    const rows = owidRowsByEntityName?.get(item.label)
+                    if (rows && rows.length > 0) return rows[0].value
+                    return 0
```

Comment:
Decide for yourself whether you like it or not:

```suggestion
                    return rows?.[0]?.value ?? 0
```